### DISCUSSION TITLE
IOne and IZero interfaces

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [dotnet/runtime#96914] [General]
Hello I am trying to write a static generic utility class.
However I ran into an issue when trying to create abs and sign methods for both numbers a vectors.

The thing is there is not interface that implements static generic getters for Zero and One and the only way I can see how to unify these operations with both unit and vector classes to implement INumber<T> for my vectors with is a really big class and is not suited at all for a vector class.

Is there a way that the C# team can split INumber<T> and INumberBase<T> One and Zero interfaces into IOne and IZero interfaces?

Infact this might it might be a good idea to implement all of INumberBase and INumber as separate interface components like this.

Here is an example of what I am trying to do...
```cs
public static class Math 
{
    public static bool InRange<T>(T value, T min, T max) where T : IComparisonOperators<T, T, bool>
    { 
        return (value >= min) && (value <= max); 
    }

    public static T Clamp<T>(T value, T min, T max)  where T : IComparisonOperators<T, T, bool>
    {
        var diff = Abs(max-min);

        if (value <  min) return min;
        if (value >= max) return max;
    }

    public static T Abs<T>(T value)  where T : struct, IComparisonOperators<T, T, bool>, IMultiplyOperators<T, int, T>
    {
        if (value < default(T)) return value * -1;
        return value;

    }
    public static T Sign<T>(T value)  where T : struct, IComparisonOperators<T, T, bool>, IMultiplyOperators<T, int, T>
    {
        if (value < default(T)) return value;
        return value;

    }
}```

**Comment by HaloFour** [comment]
That would be a [runtime](https://github.com/dotnet/runtime) request.
